src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -98,50 +97,65 @@ string DefaultMessageIdStrategy(BasicDeliverEventArgs message)
 
         static string ValueToString(object value)
         {
-            var s = value as string;
-            if (s != null)
-            {
-                return s;
-            }
-
```

Comment:
Because based on how the AMQP field tables are defined in the spec, and how the RabbitMQ client has implemented its parse of them, there are a specific set of values we can encounter here, and `string` will never be one of them.

Based on that, I see no point in keeping it around.